docs/adr/012_decouple_from_edge.md

```diff
@@ -52,77 +52,105 @@ The main difference comes to "ModuleConnectionHost", which is responsible for:
 
-IoT Edge allows you to specify those configuration variables as part of a template for [automatic deployments for single devices or at scale](https://docs.microsoft.com/en-us/azure/iot-edge/module-deployment-monitoring?view=iotedge-2020-11)
+It is decided to **identify whether LNS is running as IoT Edge Module or not by using a new "CLOUD_DEPLOYMENT" environment variable (defaults to false)**
 
-Depending on the target cloud service, you might have different ways to manage configuration in a centralized way.
+While we do not expect "IOTEDGE_APIVERSION" to change in the immediate future and, even if we are adding another environment variable, the decision is taken in order to make the identification more reliable and not depending on IoT Edge.
 
-If using Azure Kubernetes Service, a configmap to be applied to the desired cluster might work for you.
-If using Azure App Service, you might just set the environment variables in the configuration section of the application.
+As an alternative we could still use "IOTEDGE_APIVERSION" and eventually react on future changes.
 
-While these solutions are great for "static" configuration (i.e.: those values that require a restart for being changed), the LNS is coded in such a way that some configuration might be changed dynamically without having to restart the entire network server.
+#### Differentiating LNS instances in mixed-mode deployments
 
-The proposal is to **make use of "[Azure App Configuration](https://docs.microsoft.com/en-us/azure/azure-app-configuration/overview)"** to centrally manage application settings and keep [dynamic configuration using poll model](https://docs.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-aspnet-core?tabs=core5x)
+When a leaf device sends a message, its twin gets updated from the LNS with the "GatewayID" variable content, currently being set to the "IOTEDGE_DEVICEID" environment variable (defaulting to an empty string when variable is not set in the environment)
 
-### Direct method invocation
+This is the way the "Facade" Azure function has to identify and target a specific LNS instance for closing device client connections or sending C2D messages for class-c devices.
+
+The decision is to:
+
+- Prefix the **"GatewayID" with "cloud-"** for non-Edge deployments
+- Use the **"Hostname" environment variable as a default** for GatewayID field
+
+An alternative considered is to identify the difference in deployment model by querying IoT Hub for availability of a "IoT Edge" capable device and cache that result.
+
+An advantage of such alternative is the fact that there's no need to use any prefix for GatewayID which would also not tamper the observability part.
+
+A disadvantage, though, comes from the request quota limits on IoT Hub Device Registry operations which could be cause of throttling and instability for other pieces of the LNS itself.
```

Comment:
we are querying for all devices and cache them to avoid too many operations. Aren't those already in that list? Couldn't we fetch it from there?

Reply:
another alternative would be to indicate the deployment model through the API itself. Make it explicit. The parameter could be optional and default to false (edge mode), avoiding breaking existing deployments.

Reply:
regarding the second comment, I tried to write down your suggestion with f8b475b5fb543ecf0cb5d8b51a58fa92c3339859
Please feel free to add any advantage/disadvantage directly

Reply:
I didn't check what exactly we are putting into the cache, could be that we needed to refactor a bit and get the right info into the cache for the LNSs. If we would query a list of the edge devices, we could query that cache entry for the existence of the gw and if it is not in there, we know it's not an edge device. Simply a list of all edge devices that is refreshed would probably be good enough?